src/finch/tensor.py

```diff
@@ -717,12 +717,13 @@ def reshape(
 ) -> Tensor:
     if copy is False:
         raise ValueError("Unable to avoid copy during reshape.")
-    dims = [dim if dim >= 0 else jl.Colon() for dim in shape]
-    obj = jl.swizzle(x._obj, *tuple(reversed(range(1, jl.ndims(x._obj) + 1))))
-    obj = jl.reshape(obj, *reversed(dims))
-    obj = jl.swizzle(obj, *tuple(reversed(range(1, jl.ndims(obj) + 1))))
-    return Tensor(obj)
-
+    # TODO: https://github.com/finch-tensor/Finch.jl/issues/743
+    #       Revert to `jl.reshape` implementation once aforementioned
+    #       issue is solved.
+    warnings.warn("`reshape` densified the input tensor.", PerformanceWarning)
+    arr = x.todense()
+    arr = arr.reshape(shape)
+    return Tensor(arr)
```

Comment:
I would revert this and leave the job as failing for now.

Reply:
The problem here is that `reshape` function is used in multitude of Array API tests, for the input setup. It's not only `test_reshape` but a great deal of Array API tests (otherwise I'd also left it failing but now `array-api-tests` job in `pydata/sparse` is unusable).